### PR TITLE
DRTM with TrenchBoot - introduction

### PR DESCRIPTION
**Pull request by piotr-kleins**
Signed-off-by: Piotr Kleinschmidt <piotr.kleinschmidt@3mdeb.com>

### CODE REVIEW

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+---
+
+## Introduction
+
+Our company activity is to provide secure firmware and embedded systems to our
+customers. Our strategy is to use, contribute and spread open source solutions
+as much as possible. Factor, motivating to further work and showing the
+legitimacy of the adopted philosophy, is feedback in the form of satisfied
+customers, community and incoming foundations.
```
Comment (piotr-kleins) [inline blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: customers and community. We also greatly appreciate the help from foundations which want to support our projects and ideas.
@pietrushnic I suggest such change. Is it ok?

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+platforms.
+
+Final results, most likely at the end of each month, will be presented in form
+of blog articles. Everything should be clearly pointed out and explained. All
+used tools, all carried out procedures, all prepared configurations - everything
+user can inspect, revise, build, reproduce and finally use. The goal is to have
+a user-friendly and trust-worthy environment, so community can be sure about the
+quality of the solution we provide.
+
+## Summary
```
Comment (pietrushnic) [inline blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: @piotr-kleins we have to mention main TrenchBoot repository and other individuals and companies that contribute to the project Oracle, Daniel Smith, Ross, Daniel Kiper, and others. @krystian-hebel cc

Comment (krystian-hebel) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: List of all commiters/reviewers of LZ only:
```
Andrew Cooper
Daniel P. Smith
Krystian Hebel
Michał Żygowski
Piotr Król
Ross Philipson
```
In addition to those there is also one commit from Eric Snowberg to kernel, and of course Daniel Kiper reviews patches send to GRUB.

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+the entire series about **DRTM with TrenchBoot implementation**. So stay tuned
+and wait for other articles!
+
+Once again, we would like to thank **NLnet Foundation** for their support. We
+believe that our project will benefit the open-source community and will be a
+big step in development of firmware security.
+
+If you think we can help in improving the security of your firmware or you
+looking for someone who can boost your product by leveraging advanced features
+of used hardware platform, feel free to [book a call with
```
Comment (krystian-hebel) [inline blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Does it render properly with line break here?

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+## Our goal
+
+Our goal is twofold:
+1. Implement DRTM with TrenchBoot for AMD processors.
+2. Create test environment with test suites for users.
+
+Above issues will be described later. Both, as already mentioned, will be fully
+open-source and available for everybody. Solution, which we provide will be
+intended for AMD processors because it doesn't include any closed
+components.
```
[the remark this answers is not in the document]
Comment (pietrushnic) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: @krystian-hebel I would add that there 2 ACMs, ACM SINIT which can be downloaded from Intel website but is not redistributable, and ACM BIOS, which can be obtained only as OEM under CNDA with Intel.

[the remark this answers is not in the document]
Comment (krystian-hebel) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: I wasn't sure about that so decided to write it in a safe way :) Thanks for clarification

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+project's concept and describe goals, which we want to achieve. Also, this blog
+post starts a series of articles describing and verifying our progress at work.
+Thanks to that, you can observe development process and always be up-to-date
+about the project.
+
+## Our goal
+
+Our goal is twofold:
+1. Implement DRTM with TrenchBoot for AMD processors.
+2. Create test environment with test suites for users.
```
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Changed

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: I mixed both comments to a one description. Please verify if it is okay.

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+intended for AMD processors because it doesn't include any closed
+components.
+
+Now, let me briefly discuss each goal, so you can better understand the project.
+Of course, we won't focus on details and all requirements now. Those will be
+systematically updated and presented to community.
+
+#### DRTM with TrenchBoot
+
+TrenchBoot is one of the stages of boot process. It runs directly after firmware
```
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: @krystian-hebel  As it is introductory blog post, I don't delve into TrenchBoot structure too much. Let's leave technical details for future articles.

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+(coreboot in our case) execution ended. It is a technology which measures and
+verifies running environment. To obtain that, TrenchBoot utilizes **Dynamic Root
+of Trust for Measurements** (DRTM). It is very complex topic and understanding
+it could take too much time. What you need to know (or trust us about) is that
+those measurements are done for given piece of code (given piece of system) and
+are stored in **Trusted Platform Module** (TPM) special registers. It gives
+possibility to automatically verify them against corruption and eventually
+proceed or stop further booting. Also, the user has possibility to check if
```
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+measurements are valid. It can be done from OS level using dedicated tools. So,
+as you can see, TrenchBoot is critical piece of firmware, if you want to make it
```
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md (hunk 2 above)
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Added in summary section

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md (hunk 3 above)
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Revert to original formatting

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+secure. Also, its implementation is fully open-source, so everyone can inspect
+it and contribute.
+
+#### Test environment
+
+All requirements of project are automatically or manually validate after
```
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+TrenchBoot is one of the stages of boot process. It runs directly after firmware
+(coreboot in our case) execution ended. It is a technology which measures and
+verifies running environment. To obtain that, TrenchBoot utilizes **Dynamic Root
+of Trust for Measurements** (DRTM). It is very complex topic and understanding
+it could take too much time. What you need to know (or trust us about) is that
+those measurements are done for given piece of code (given piece of system) and
+are stored in **Trusted Platform Module** (TPM) special registers. It gives
+possibility to automatically verify them against corruption and eventually
+proceed or stop further booting. Also, the user has possibility to check if
+measurements are valid. It can be done from OS level using dedicated tools. So,
```
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+Of course, we won't focus on details and all requirements now. Those will be
+systematically updated and presented to community.
+
+#### DRTM with TrenchBoot
+
+TrenchBoot is one of the stages of boot process. It runs directly after firmware
+(coreboot in our case) execution ended. It is a technology which measures and
+verifies running environment. To obtain that, TrenchBoot utilizes **Dynamic Root
+of Trust for Measurements** (DRTM). It is very complex topic and understanding
+it could take too much time. What you need to know (or trust us about) is that
```
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+about the project.
+
+## Our goal
+
+Our goal is twofold:
+1. Implement DRTM with TrenchBoot for AMD processors.
+2. Create test environment with test suites for users.
+
+Above issues will be described later. Both, as already mentioned, will be fully
+open-source and available for everybody. Solution, which we provide will be
```
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+  - coreboot
+categories:
+  - Firmware
+  - Security
+
+---
+
+## Introduction
+
+Our company activity is to provide secure firmware and embedded systems to our
```
[the remark this answers is not in the document]
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 14]
```diff
@@ -0,0 +1,121 @@
+---
+title: Open Source DRTM with TrenchBoot for AMD processors. Introduction.
+abstract: This article starts an entire series of articles related to title
+          project. Today, you will find out why we have started such project and
```
Comment (miczyg1) [inline blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: As in other post I suggest to not use time-related words. Please use something like: `by reading this blog post, you will find out ...`

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+categories:
+  - Firmware
+  - Security
+
+---
+
+## Introduction
+
+Our company's activity is to provide secure firmware and embedded systems to our
+customers. Our strategy is to use, contribute and spread open source solutions
```
Comment (miczyg1) [inline blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]:
```suggestion
customers. 3mdeb strategy is to use, contribute and spread open source solutions
```
Too many `our`

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+Our goal is twofold:
+1. Implement DRTM with TrenchBoot for AMD processors.
+2. Create a test environment with test suites for the community.
+
+Above issues will be described later. Both, as already mentioned, will be fully
+open-source and available for everybody. Solution which we provide will be
+intended for AMD processors because it doesn't include any closed software
+components. To do something similar on Intel processors, components called ACM
+(Authenticated Code Module) are required. More precisely, first is ACM SINIT
+which be downloaded from Intel website, but is not redistributable. Second is
```
Comment (miczyg1) [inline blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]:
```suggestion
which can be downloaded from Intel website, but is not redistributable. Second is
```

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+
+Now, let me briefly discuss each goal, so you can better understand the project.
+Of course, we won't focus on details and all requirements now. Those will be
+systematically updated and presented to community.
+
+#### DRTM with TrenchBoot
+
+[TrenchBoot](https://github.com/TrenchBoot) *is a framework that allows
+individuals and projects to build security engines to perform launch integrity
+actions for their systems.* In the other words, it gives a tools to create the
```
Comment (miczyg1) [inline blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]:
```suggestion
actions for their systems.* In other words, it provides tools to create the
```

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md (hunk 14 above)
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md (hunk 15 above)
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md (hunk 16 above)
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: done

File: blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md (hunk 17 above)
Comment (piotr-kleins) [reply blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md]: Done

**Pull request merged**